app/src/main/java/ar/com/wolox/android/trainingFS/network/youtube/YoutubeAdapter.kt

```diff
@@ -0,0 +1,85 @@
+package ar.com.wolox.android.trainingFS.network.youtube
+
+import android.content.Context
+import android.util.Log
+import ar.com.wolox.android.trainingFS.model.youtube.YoutubeResponse
+import com.google.gson.GsonBuilder
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import javax.inject.Inject
+
+class YoutubeAdapter @Inject constructor(val context: Context) {
+
+    fun getSongs(query: String, pageToken: String, listener: IYoutubeAdapterListener) {
+        val retrofit: Retrofit = Retrofit.Builder()
+                                    .baseUrl(BASE_URL)
+                                    .addConverterFactory(GsonConverterFactory.create())
+                                    .build()
+
+        val api = retrofit.create(YoutubeService::class.java)
+        val body: HashMap<String, String> = hashMapOf()
+        body[KEY_PART] = VALUE_PART
+        body[KEY_QUERY] = query
+        body[KEY_API] = VALUE_API
+        body[KEY_TYPE] = VALUE_TYPE
+        body[KEY_MAX] = VALUE_MAX
+
+        if (pageToken.isNotEmpty()) {
+            body[KEY_TOKEN] = pageToken
+        }
+
+        val call = api.searchSongs(body)
+        call.enqueue(object : Callback<YoutubeResponse> {
+            override fun onResponse(call: Call<YoutubeResponse>, response: Response<YoutubeResponse>) {
+
+                var result = response.body()
```

Comment:
I used "var result = response.body()" instead "var result = response.body() ?: getSampleFromAssets()" because the login inside the near if is only for test (dummy simulation). When the api key works without problems, the logic needs to be deleted

Reply:
Yes, I checked that. I meant: 
```
val result = if (response.isSuccessful) response.body() else getSampleFromAssets()
```

And if you want to leave the Log:
```
val result = 
        if (response.isSuccessful) {
                 response.body() 
        } else {
                 Log.e("YTSimulation", "ResponseError: " + response.errorBody()?.string())
                 getSampleFromAssets()
        }
```

Reply:
ahhhh ok, i did not understand the first time. my bad!